menas/ui/components/hdfs/HDFSBrowser.js

```diff
@@ -200,25 +196,22 @@ sap.ui.define([], function() {
         var items = that._tree.getItems();
         for ( var i in items) {
           var bindingPath = items[i].getBindingContextPath();
-          var hdfsPath = that._model.getProperty(bindingPath).path
-          if (hdfsPath === sNewPath) {
-            try{
-              that._tree.expand(parseInt(i))
-            } catch (e) {
-              // some weirdness if theres no such child
-            }
+          let oItem = that._model.getProperty(bindingPath);
+          var hdfsPath = oItem["path"];
+          if (hdfsPath === sNewPath && oItem["children"] !== null && 
+              (sPath === "/" || hdfsPath !== sPath)) {
+            let index = parseInt(i);
+            that._tree.expand(index);
           }
           // also select the correct list item
           if (hdfsPath === sPath) {
             items[i].setSelected(true);
           }
         }
-        if(rev.length === 0) {
-          // this binding seems weird on these nested models
+        if(rev.length > 0) fnHelper(sNewPath, sModelPath, rev.reverse());
```

Comment:
```suggestion
        if(rev.length > 0) {
          fnHelper(sNewPath, sModelPath, rev.reverse());
        } else {
```
Let's follow the agreed on style for multi-line `if-else` and add curly braces